api/organisations/subscriptions/dataclasses.py

```diff
@@ -0,0 +1,57 @@
+import typing
+
+
+class BaseSubscriptionMetadata:
+    def __init__(
+        self,
+        seats: int = 0,
+        api_calls: int = 0,
+        projects: typing.Optional[int] = None,
+        payment_source: str = None,
+    ):
+        self.seats = seats
+        self.api_calls = api_calls
+        self.projects = projects
+        self.payment_source = payment_source
+
+    def __add__(self, other: "BaseSubscriptionMetadata"):
+        if self.payment_source != other.payment_source:
+            raise TypeError(
+                "Cannot add SubscriptionMetadata from multiple payment sources."
+            )
+
+        # TODO: test this logic and confirm that it's correct
```

Comment:
Are we planning to resolve this TODO in next PR?

Reply:
Good catch - no, I just forgot to do it. I have added it now.